contrib/message-logging/message-logging-parser.py

```diff
@@ -0,0 +1,127 @@
+            val = getattr(obj, slot, None)
+            if slot in HASH_INTS and isinstance(val, int):
+                ret[slot] = ser_uint256(val).hex()
+            elif slot in HASH_INT_VECTORS and isinstance(val[0], int):
+                ret[slot] = [ser_uint256(a).hex() for a in val]
+            else:
+                ret[slot] = to_jsonable(val)
+        return ret
+    elif isinstance(obj, list):
+        return [to_jsonable(a) for a in obj]
+    elif isinstance(obj, bytes):
+        return obj.hex()
+    else:
+        return obj
+
+
+def process_file(path: str, messages: List[Any], recv: bool) -> None:
+    with open(path, 'rb') as f_in:
+        while True:
+            tmp_header_raw = f_in.read(TIME_SIZE + LENGTH_SIZE + MSGTYPE_SIZE)
+            if not tmp_header_raw:
+                break
+            tmp_header = BytesIO(tmp_header_raw)
+            time = int.from_bytes(tmp_header.read(TIME_SIZE), "little")      # type: int
+            msgtype = tmp_header.read(MSGTYPE_SIZE).split(b'\x00', 1)[0]     # type: bytes
+            length = int.from_bytes(tmp_header.read(LENGTH_SIZE), "little")  # type: int
+            if msgtype not in MESSAGEMAP:
+                # For now just skip unrecognized messages
+                f_in.read(length)
+                continue
+            payload_start_pos = f_in.tell()
+            msg = MESSAGEMAP[msgtype]()
+            msg.deserialize(f_in)
+            payload_length = f_in.tell() - payload_start_pos
+            assert_equal(length, payload_length)
+            msg_dict = {}
+            msg_dict["msgtype"] = getattr(msg, "msgtype", None).decode()
+            msg_dict["direction"] = "recv" if recv else "sent"
+            msg_dict["time"] = time
+            msg_dict["size"] = length   # "size" is less readable here, but more readable in the output
+            if length:
+                msg_dict["body"] = to_jsonable(msg)
+            messages.append(msg_dict)
+
+
+def main():
+    """Main"""
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        epilog="EXAMPLE \n\t{0} -o out.json <data-dir>/message_logging/**/*.dat".format(sys.argv[0]),
+        formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument(
+        "logpaths", nargs='+',
+        help="binary message dump files to parse.")
+    parser.add_argument("-o", "--output", help="output file.  If unset print to stdout")
+    args = parser.parse_args()
+    logpaths = [Path.cwd() / Path(logpath) for logpath in args.logpaths]
+    output = Path.cwd() / Path(args.output) if args.output else False
+
+    messages = []   # type: List[Any]
+    for log in logpaths:
+        process_file(str(log), messages, "recv" in log.stem)
+
+    messages.sort(key=lambda msg: msg['time'])
+
+    jsonrep = json.dumps(messages)
+    if output:
+        with open(str(output), 'w+', encoding="utf8") as f_out:
+            f_out.write(jsonrep)
+    else:
+        print(jsonrep)
```

Comment:
There may be a better way to do it, but I found the stdout to be much more readable with:
```diff
 import os
+import pprint
 import sys
...
-        print(jsonrep)
+        pprint.pprint(jsonrep)
```

Reply:
or just piping trough `jq`, gives you colors too